docs/token-contracts/fa12/4-fa12-archetype.md

```diff
@@ -0,0 +1,447 @@
+---
+id: 4-fa12-archetype
+title: Deploy a Verified Implementation of FA1.2 Contract
+sidebar_label: FA1.2 Archetype
+---
+
+## Introduction
+This article presents the procedure to deploy a verified implementation of the FA1.2 contract.
+
+[Program verification](https://en.wikipedia.org/wiki/Formal_verification) is a technique to mathematically prove that a program has the required properties. It provides a high level of confidence that the contract executes correctly according to its specification.
+
+The formal properties of the FA1.2 contract are presented [below](#formal-properties).
+
+The contract and its formal specification are written in the [Archetype language](https://archetype-lang.org/), a high-level domain-specific language dedicated to the development of verified smart contracts for the Tezos blockchain.
+
+For verification purposes, Archetype translates the contract to the Why3 language; [Why3](http://why3.lri.fr/) is a generic program verification platform; it generates proof obligations and calls [SMT](https://en.wikipedia.org/wiki/Satisfiability_modulo_theories) solvers to solve them.
+
+## Generate Michelson
+The Archetype source code is available [here](https://github.com/edukera/archetype-lang/blob/0a5ad0832709ac102a14534f22d4f94cb185866d/contracts/fa12.arl).
+
+> SHA256 (fa12.arl) = d38260e5f55e630880ecd5d5ed32eb91988d2f34b43a7c031275d8f8a8715184
+
+The following command generates the Michelson version of the contract in the `fa12.tz` file:
+```
+$ archetype fa12.arl > fa12.tz
+```
+
+> Install [Archetype](https://docs.archetype-lang.org/getting-started-1#installation).
```

Comment:
The naive approach failed:

```bash
$ opam install archetype
The following dependencies couldn't be met:
  - archetype → ocaml >= 4.07.0
      base of this switch (use `--unlock-base' to force)

No solution found, exiting
```

My solution:
```bash
mkdir archetype
cd archetype
opam switch create . 4.07.0
eval $(opam env) 
opam install archetype --yes
```

But this installed the wrong version of Archetype: `0.1.10`.

```bash
$ opam upgrade archetype 
Already up-to-date.
Nothing to do.
```

Running `opam update` then `opam upgrade archetype` installed `archetype 1.2.0`.

Reply:
What happened is that you were in a switch with a version of ocaml less than 4.07.0
In that case you need to create another switch as you did.
Indeed the `opam update` cmd is necessary to retrieve last packages version information.